### PR TITLE
Add highlight to multi-select sidebar on hover of entity

### LINKED ISSUE #2949: With multiple features selected, indicate which one is mouse-hovered in the sidebar multiselect list
Edit: The remaining task is highlighting the list item when hovering the selected feature on the map, which is discussed and agreed on starting here: https://github.com/openstreetmap/iD/issues/2949#issuecomment-431540750
Original issue:

Identification of items in the multiselect list should be improved:

When the mouse is over an item of the list, selection halos of all other selected elements should be disabled temporarily.

This will likely require similar software infrastructure as #2946 , and is a prerequisite for some further potential enhancements of iD.

### PR DESCRIPTION
**Pull request by jgscherber**
This is to resolve #2949 and is a WIP

This is my first time working on this project (and first time working in a large JS codebase) so please don't hold back on style/design criticisms. I want to thank you for the Architecture readme that you have. It really helped with initially finding my way around.

Here's a demo of the highlight
highlight_demo

I ran into a couple issues while working on it
* After adding the `hoverModeSelect` callback, performing a click-and-drag results in iD dropping out of "select" mode. Any ideas what's causing that?
* The discussion in #2949 says the resolution to this issue should also have the sidebar scroll into view the element in the list that's being highlighted - how can I check if an element is visible in the viewport?
* This change adds a class `.hover` to perform the styling which duplicates the pseudo-class `:hover` -- is it better for the `:hover` pseudo-class be removed and only styled using the `.hover` class

### CODE REVIEW

**Comment by quincylvania** [comment]
@jgscherber Thanks for working on this! I'll take a look and give you feedback this week.

File: modules/ui/sidebar.js [hunk 1]
```diff
@@ -113,6 +113,24 @@ export function uiSidebar(context) {
             .append('div')
             .attr('class', 'inspector-hidden inspector-wrap fr');
 
+        var hoverModeSelect = function hoverModeSelect(datum) {
```
Comment (quincylvania) [inline modules/ui/sidebar.js:156]: No need to name this twice, you can pick either style.

File: modules/ui/sidebar.js [hunk 2]
```diff
@@ -113,6 +113,24 @@ export function uiSidebar(context) {
...
+
+            if (context.mode().selectedIDs().length > 1 
+                && (datum instanceof osmEntity)){
+                
+                var elements = context.container().selectAll('.feature-list-item')
+                    .filter(function (node) {
+                        return node.id === datum.id;
+                });
+                
+                if (elements.size() > 0){
```
Comment (quincylvania) [inline modules/ui/sidebar.js:167]: This works fine but there's a `.empty()` function that's even easier.

**Comment by quincylvania** [comment]
@jgscherber Looks pretty good! We could merge as-is but I'll give you a chance to take another look if you want. Let me know if you have any questions.

> After adding the hoverModeSelect callback, performing a click-and-drag results in iD dropping out of "select" mode. Any ideas what's causing that?

This was a bug with other changes I'd been making in the `develop` branch. If you remerge this should be fixed.

> The discussion in #2949 says the resolution to this issue should also have the sidebar scroll into view the element in the list that's being highlighted - how can I check if an element is visible in the viewport?

I think we might actually live without this. I feel like it'd be disorienting more often than not. But to answer your question you can compare the list item element's position to [`scrollTop`](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTop) of the inspector.

> This change adds a class .hover to perform the styling which duplicates the pseudo-class :hover -- is it better for the :hover pseudo-class be removed and only styled using the .hover class

Nah, there's no harm in two separate classes. I had to move all the `:hover` declarations to media queries anyway for #7432.

**Comment by jgscherber** [comment]
I'm planning on changing your two suggestions. I'll fix my rebase also

**Pull request closed without merging**